opensrp-app/src/main/java/org/smartregister/repository/ClientFormRepository.java

```diff
@@ -0,0 +1,153 @@
+
+import org.apache.commons.lang3.StringUtils;
+import org.smartregister.domain.ClientForm;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+import timber.log.Timber;
+
+/**
+ * Created by cozej4 on 2020-04-06.
+ *
+ * @author cozej4 https://github.com/cozej4
+ */
+public class ClientFormRepository extends BaseRepository {
+
+    protected static final String ID = "id";
+    protected static final String VERSION = "version";
+    protected static final String IDENTIFIER = "identifier";
+    protected static final String MODULE = "module";
+    protected static final String JSON = "json";
+    protected static final String JURISDICTION = "jurisdiction";
+    protected static final String LABEL = "label";
+    protected static final String IS_NEW = "is_new";
+    protected static final String ACTIVE = "active";
+    protected static final String CREATED_AT = "created_at";
+    protected static final String CLIENT_FORM_TABLE = "client_form";
+    public static final String CREATE_CLIENT_FORM_TABLE =
+            "CREATE TABLE " + CLIENT_FORM_TABLE + " (" +
+                    ID + " VARCHAR NOT NULL PRIMARY KEY," +
+                    VERSION + " VARCHAR , " +
+                    IDENTIFIER + " VARCHAR , " +
+                    MODULE + " VARCHAR , " +
+                    JSON + " VARCHAR , " +
+                    JURISDICTION + " VARCHAR , " +
+                    LABEL + " VARCHAR , " +
+                    IS_NEW + " INTEGER , " +
+                    ACTIVE + " INTEGER  NOT NULL, " +
+                    CREATED_AT + " VARCHAR NOT NULL ) ";
+    protected static final String[] COLUMNS = new String[]{ID, VERSION, IDENTIFIER, MODULE, JSON, JURISDICTION, LABEL, IS_NEW, ACTIVE, CREATED_AT};
+    private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss", Locale.ENGLISH);
+    private static final String CREATE_CLIENT_FORM_IDENTIFIER_INDEX = "CREATE INDEX "
+            + CLIENT_FORM_TABLE + "_" + IDENTIFIER + "_ind ON " + CLIENT_FORM_TABLE + "(" + IDENTIFIER + ")";
+    private static final String CREATE_CLIENT_FORM_IS_ACTIVE_INDEX = "CREATE INDEX "
+            + CLIENT_FORM_TABLE + "_" + ACTIVE + "_ind ON " + CLIENT_FORM_TABLE + "(" + ACTIVE + ")";
+
+    public static void createTable(SQLiteDatabase database) {
+        database.execSQL(CREATE_CLIENT_FORM_TABLE);
+        database.execSQL(CREATE_CLIENT_FORM_IDENTIFIER_INDEX);
+        database.execSQL(CREATE_CLIENT_FORM_IS_ACTIVE_INDEX);
+    }
+
+    protected String getClientFormTableName() {
+        return CLIENT_FORM_TABLE;
+    }
+
+    public void addOrUpdate(ClientForm clientForm) {
+        if (StringUtils.isBlank(clientForm.getId()))
+            throw new IllegalArgumentException("id not provided");
+        ContentValues contentValues = new ContentValues();
+        contentValues.put(ID, clientForm.getId());
```

Comment:
You will need to test out this out. I think the ID provided by the server is int and not string

Reply:
Okay cool, Let me change it to an int